service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOControllerTest.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.util.UUID.fromString;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftCMOController.class)
+@OverrideAutoConfiguration(enabled = true)
+@SuppressWarnings("unchecked")
+class DraftCMOControllerTest {
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+
+    @Autowired
+    private DraftCMOService draftCMOService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private final LocalDate date = LocalDate.now();
+    private final List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofLocalizedDate(
+        FormatStyle.MEDIUM).localizedBy(Locale.UK);
+
+    @Test
+    void aboutToStartCallbackShouldPopulateHearingDatesList() throws Exception {
+        Map<String, Object> data = ImmutableMap.of("hearingDetails", hearingDetails);
+
+        List<String> expected = Arrays.asList(
+            date.plusDays(5).format(dateTimeFormatter),
+            date.plusDays(2).format(dateTimeFormatter),
+            date.format(dateTimeFormatter));
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = getResponse(data, "about-to-start");
+
+        assertThat(getHearingDates(callbackResponse)).isEqualTo(expected);
+    }
+
+    @Test
+    void aboutToSubmitShouldPopulateHiddenHearingDateField() throws Exception {
+        List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+
+        DynamicList dynamicHearingDates = draftCMOService.buildDynamicListFromHearingDetails(hearingDetails);
+
+        dynamicHearingDates
+            .setValue(
+                DynamicListElement.builder()
+                    .code(fromString("b15eb00f-e151-47f2-8e5f-374cc6fc2657"))
+                    .label(date.plusDays(5).toString())
+                    .build());
+
+        Map<String, Object> data = ImmutableMap.of("cmoHearingDateList", dynamicHearingDates);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = getResponse(data, "about-to-submit");
+
+        assertThat(callbackResponse.getData()).doesNotContainKey("cmoHearingDateList");
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(caseData.getCaseManagementOrder().getHearingDate())
+            .isEqualTo(date.plusDays(5).toString());
+    }
```

Comment:
can you also assert that the ID is here? The ID is arguably more important than the date as this is your current link between the list and the CMO

Reply:
done